src/coreComponents/physicsSolvers/fluidFlow/ReactiveCompositionalMultiphaseOBL.cpp

```diff
@@ -1218,9 +1223,63 @@ void ReactiveCompositionalMultiphaseOBL::applyDirichletBC( real64 const time,
 // to be changed into enforceOBLLimits - to chop all primary variables to be within OBL discretization space
 void ReactiveCompositionalMultiphaseOBL::chopPrimaryVariablesToOBLLimits( DomainPartition & domain )
 {
-  GEOSX_UNUSED_VAR( domain );
+  real64 const eps = LvArray::NumericLimits< real64 >::epsilon;
+  integer const numComp = m_numComponents;
+
+  forDiscretizationOnMeshTargets( domain.getMeshBodies(), [&]( string const &,
+                                                               MeshLevel & mesh,
+                                                               arrayView1d< string const > const & regionNames )
+  {
+    mesh.getElemManager().forElementSubRegions( regionNames,
+                                                [&]( localIndex const,
+                                                     ElementSubRegionBase & subRegion )
+    {
+      arrayView2d< real64, compflow::USD_COMP > const compFrac =
+        subRegion.template getExtrinsicData< extrinsicMeshData::flow::globalCompFraction >();
+
+      forAll< parallelDevicePolicy<> >( subRegion.size(), [=] GEOSX_HOST_DEVICE ( localIndex const ei )
+      {
+        bool isScalingRequired = false;
+
+        // Step 1: chop the component fractions between 0 and 1
```

Comment:
Why do you need `eps` for each `compFrac`?
Can't it be 1 or 0?

Reply:
Here, what is implemented follows exactly what is in DARTS, which is why I would prefer to keep it like this. This type of chopping after the Newton update does not have any theoretical foundation and is often full of tolerances and buffers... but it works quite well. 

One thing to note is that the `eps` ensures that each component always exists in the system (even just as a trace), which may have an impact on the linear solver behavior... but we have not tested that yet with OBL.

Reply:
ok; maybe comment in the code that you're using the recipe from DARTS.
Does it make sense?

Reply:
Yes, I added this in the comment before the chopping algorithm